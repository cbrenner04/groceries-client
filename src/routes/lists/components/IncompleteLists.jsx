import React from 'react';

import AcceptedLists from './AcceptedLists';
import TitlePopover from '../../../components/TitlePopover';

const IncompleteLists = (props) => (
  <AcceptedLists
    title={
      <TitlePopover
        title="Incomplete"
        message="These are lists you've created or you've accepted an invitation from someone else."
      />
    }
    completed={false}
    {...props}
  />
);

export default IncompleteLists;
