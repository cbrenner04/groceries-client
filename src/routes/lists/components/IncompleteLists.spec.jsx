import React from 'react';
import { render } from '@testing-library/react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';

import IncompleteLists from './IncompleteLists';

describe('IncompleteLists', () => {
  let props;

  const renderIncompleteLists = (p) => {
    const history = createMemoryHistory();
    return render(
      <Router history={history}>
        <IncompleteLists {...p} />
      </Router>,
    );
  };

  beforeEach(() => {
    props = {
      history: {
        push: jest.fn(),
      },
      userId: 1,
      incompleteLists: [
        {
          id: 1,
          name: 'foo',
          type: 'GroceryList',
          owner_id: 1,
          created_at: new Date('05/27/2020').toISOString(),
          completed: false,
          users_list_id: 1,
          refreshed: false,
        },
      ],
      setIncompleteLists: jest.fn(),
      completedLists: [
        {
          id: 1,
          name: 'foo',
          type: 'GroceryList',
          owner_id: 1,
          created_at: new Date('05/27/2020').toISOString(),
          completed: true,
          users_list_id: 1,
          refreshed: false,
        },
      ],
      setCompletedLists: jest.fn(),
      currentUserPermissions: {
        1: 'write',
      },
      setCurrentUserPermissions: jest.fn(),
    };
  });

  it('renders', () => {
    const { container } = renderIncompleteLists(props);

    expect(container).toMatchSnapshot();
  });
});
