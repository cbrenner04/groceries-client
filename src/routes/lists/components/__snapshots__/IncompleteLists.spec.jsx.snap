// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IncompleteLists renders 1`] = `
<div>
  <div
    class="mb-4"
  >
    <div
      class="clearfix"
    >
      <div
        class="mx-auto float-left"
      >
        <div
          class="row"
          style="margin: 0px;"
        >
          <h2
            class="text-capitalize"
          >
            Incomplete
          </h2>
          <i
            class="far fa-question-circle float-right text-secondary"
            data-test-id="Incomplete-popover"
          />
        </div>
      </div>
      <button
        class="mx-auto float-right btn btn-link"
        type="button"
      >
        
         Select
      </button>
    </div>
    <div
      class="list-group"
    >
      <div
        class="accepted-list list-group-item"
        data-test-class="incomplete-list"
        data-test-id="list-1"
        style="display: block;"
      >
        <div
          class="row"
        >
          <div
            class="col-sm-12 col-12"
          >
            <div
              class="row"
            >
              <div
                class="pt-1 col-md-6"
              >
                <a
                  class="router-link"
                  href="/lists/1"
                >
                  <h5
                    class="mb-1"
                  >
                    <i
                      class="fa fa-shopping-bag text-info mr-3"
                    />
                    foo
                  </h5>
                </a>
              </div>
              <div
                class="pt-1 col-md-4"
              >
                <small
                  class="text-muted"
                >
                  May 27 2020, 12:00:00 am
                </small>
              </div>
              <div
                class="col-md-2"
              >
                <div
                  class="float-right btn-group"
                  role="group"
                >
                  <button
                    class="p-0 mr-3 btn btn-link"
                    data-test-id="incomplete-list-complete"
                    style="pointer-events: auto; opacity: 1;"
                    type="button"
                  >
                    <i
                      class="fa fa-check fa-2x text-success"
                    />
                  </button>
                  <a
                    class="p-0 mr-3"
                    data-test-id="incomplete-list-share"
                    href="/lists/1/users_lists"
                    style="pointer-events: auto; opacity: 1;"
                  >
                    <i
                      class="fa fa-users fa-2x text-primary"
                    />
                  </a>
                  <a
                    class="p-0 mr-3"
                    data-test-id="incomplete-list-edit"
                    href="/lists/1/edit"
                    style="pointer-events: auto; opacity: 1;"
                  >
                    <i
                      class="fa fa-edit fa-2x text-warning"
                    />
                  </a>
                  <button
                    class="p-0 btn btn-link"
                    data-test-id="incomplete-list-trash"
                    type="button"
                  >
                    <i
                      class="fa fa-trash fa-2x text-danger"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
